packages/calcite-components/src/components/sheet/sheet.tsx

```diff
@@ -0,0 +1,365 @@
+  @Prop({ mutable: true, reflect: true }) open = false;
+
+  @Watch("open")
+  async toggleSheet(value: boolean): Promise<void> {
+    if (this.ignoreOpenChange) {
+      return;
+    }
+
+    onToggleOpenCloseComponent(this);
+    if (value) {
+      this.openSheet();
+    } else {
+      this.closeSheet();
+    }
+  }
+
+  /**
+   * We use an internal property to handle styles for when a modal is actually opened, not just when the open attribute is applied. This is a property because we need to apply styles to the host element and to keep the styles present while beforeClose is .
+   *
+   * @internal.
+   */
+  @Prop({ mutable: true, reflect: true }) opened = false;
+
+  /** When `true`, disables the closing of the component when clicked outside. */
+  @Prop({ reflect: true }) outsideCloseDisabled = false;
+
+  /** Determines where the component will be positioned. */
+  @Prop({ reflect: true }) position: LogicalFlowPosition = "inline-start";
+
+  /**
+   * This internal property, managed by a containing calcite-shell, is used
+   * to inform the component if special configuration or styles are needed
+   *
+   * @internal
+   */
+  @Prop() slottedInShell: boolean;
+
+  /**
+   * When `position` is `"inline-start"` or `"inline-end"`, specifies the width of the component.
+   */
+  @Prop({ reflect: true }) widthScale: Scale = "m";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  async componentWillLoad(): Promise<void> {
+    setUpLoadableComponent(this);
+    // when sheet initially renders, if active was set we need to open as watcher doesn't fire
+    if (this.open) {
+      onToggleOpenCloseComponent(this);
+      requestAnimationFrame(() => this.openSheet());
+    }
+  }
+
+  componentDidLoad(): void {
+    setComponentLoaded(this);
+  }
+
+  connectedCallback(): void {
+    this.mutationObserver?.observe(this.el, { childList: true, subtree: true });
+    connectFocusTrap(this);
+  }
+
+  disconnectedCallback(): void {
+    this.removeOverflowHiddenClass();
+    this.mutationObserver?.disconnect();
+    deactivateFocusTrap(this);
+    this.slottedInShell = false;
+  }
+
+  render(): VNode {
+    const dir = getElementDir(this.el);
+    return (
+      <Host
+        aria-describedby={this.contentId}
+        aria-label={this.label}
+        aria-modal="true"
+        role="dialog"
+      >
+        <div
+          class={{
+            [CSS.container]: true,
+            [CSS.containerOpen]: this.opened,
+            [CSS.slottedInShell]: this.slottedInShell,
+            [CSS_UTILITY.rtl]: dir === "rtl",
+          }}
+        >
+          <calcite-scrim class={CSS.scrim} onClick={this.handleOutsideClose} />
+          <div
+            class={{
+              [CSS.content]: true,
+            }}
+            // eslint-disable-next-line react/jsx-sort-props
```

Comment:
Can you add the explainer from https://github.com/Esri/calcite-design-system/pull/7584? 

> // eslint-disable-next-line react/jsx-sort-props -- ref should be last so node attrs/props are in sync (see https://github.com/Esri/calcite-design-system/pull/6530)

Reply:
applies to modal as well.